epm/pkg/utils/cmsg.go

```diff
@@ -0,0 +1,71 @@
+package utils
+
+import(
+	"net"
+	"syscall"
+)
+
+func RecvFd(socksPath string) (int, error) {
+        var fds []int
+
+        laddr, err := net.ResolveUnixAddr("unix", socksPath)
+        if err != nil {
+                panic(err)
+        }
+
+        l, err := net.ListenUnix("unix", laddr)
+        if err != nil {
+                panic(err)
+        }
+
+        conn, err := l.AcceptUnix()
+        if err != nil {
+                panic(err)
+        }
+        // Part msg into both datas
+        buf := make([]byte, 32)
+        oob := make([]byte, 32)
+        _, oobn, _, _, err := conn.ReadMsgUnix(buf, oob)
+        if err != nil {
+                panic(err)
+        }
+        // ParseSocketControlMessage Array
+        scms, err := syscall.ParseSocketControlMessage(oob[:oobn])
+        if err != nil {
+                panic(err)
+        }
+        if len(scms) > 0 {
+                // Get UnixRights from SocketControlMessage
+                fds, err = syscall.ParseUnixRights(&(scms[0]))
+                if err != nil {
+                        panic(err)
+                }
+        }
+
+        err = conn.Close()
+        if err != nil {
+                panic(err)
+        }
+	// Delete original unixSock by unlink
+        syscall.Unlink(socksPath)
+	return fds[0], err
+}
+
+func SendFd(socksPath string, fd int) error {
```

Comment:
ditto. Refer to https://github.com/opencontainers/runc/blob/master/libcontainer/utils/cmsg.go#L87

Reply:
ok, I will check it. Thanks